client/http/errors.go

```diff
@@ -0,0 +1,15 @@
+package http
+
+import "github.com/go-resty/resty/v2"
+
+type FailedResponseError struct {
+	res *resty.Response
+}
+
+func (e *FailedResponseError) Error() string {
+	return e.res.Status() + ": " + string(e.res.Body())
+}
```

Comment:
When is this ever used? Is `Error` called anywhere along the way when displaying errors?

Reply:
In general for a struct to be considered an error, it must implement Error().